UDPClientGUI/ViewModels/ShellViewModel.cs

```diff
@@ -79,10 +84,20 @@ public void Disconnect(string iP, string port)
             _eventAggregator.PublishOnBackgroundThread(new ClientCommandEvent(ClientCommandEvent.Command.Disconnect, IP, Port));
         }
 
+        public void Login()
+        {
+            IWindowManager manager = new WindowManager();
+            manager.ShowDialog(_loginViewModel, null, null);
+        }
+
         public bool CanSend(string message)
         {
-            if (ConnectionStatus == "Connected") return !(message is null || message == "");
-            return false;
+            //return (ConnectionStatus == "Connected") ? (!string.IsNullOrEmpty(message)) : false;
+            var Connected = (ConnectionStatus == "Connected");
+            var EmptyMessage = string.IsNullOrEmpty(message);
+            return (Connected && !EmptyMessage);
+            //if (ConnectionStatus == "Connected") return !(message is null || message == "");
+            //return false;
```

Comment:
Please remove commented out code.

lower case first letter of local variable names.

var connected = ....
var emptyMessage ==